sdk/core/Azure.Core.Expressions.DataFactory/src/DataFactorySecretBaseDefinition.cs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+namespace Azure.Core.Expressions.DataFactory
+{
+    /// <summary>
+    /// The base definition of a secret type.
+    /// Please note <see cref="DataFactorySecretBaseDefinition"/> is the base class. According to the scenario, a derived class of the base class might need to be assigned here, or this property needs to be casted to one of the possible derived classes.
+    /// The available derived classes include <see cref="DataFactorySecretString"/> and <see cref="DataFactoryKeyVaultSecretReference"/>.
+    /// </summary>
+    [TypeReferenceType]
+    public abstract partial class DataFactorySecretBaseDefinition
```

Comment:
This type was extracted from the Azure.ResourceManager.DataFactory library. The plan is to have code gen use the type from here by leveraging the TypeReferenceType attribute.